### PR TITLE
rust-analyzer: enable check by default

### PR DESCRIPTION
**Pull request by oxalica**
###### Motivation for this change

We have rustc 1.53.0 in nixpkgs and the tests should pass now.
Currently only the crate `rust-analyzer` is checked. Other dependent crates are not since it will cause too many crate rebuilds.

Tested via the vscode extension.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 135610` at 131f5c3b run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>rust-analyzer</li>
    <li>rust-analyzer-unwrapped</li>
    <li>vscode-extensions.matklad.rust-analyzer</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/development/tools/rust/rust-analyzer/default.nix:20:5:
```
   |
20 |     ./ignore-git-and-rustfmt-tests.patch
   |     ^
```
</li>
  </ul>
</details>

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 135610` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>rust-analyzer</li>
    <li>rust-analyzer-unwrapped</li>
    <li>vscode-extensions.matklad.rust-analyzer</li>
  </ul>
</details>

File: pkgs/development/tools/rust/rust-analyzer/default.nix [hunk 1]
```diff
@@ -17,9 +16,14 @@ rustPlatform.buildRustPackage rec {
     sha256 = "sha256-6Tbgy77Essi3Hyd5kdJ7JJbx7RuFZQWURfRrpScvPPQ=";
   };
 
+  patches = [
```
Comment (happysalada) [inline pkgs/development/tools/rust/rust-analyzer/default.nix:19]: I would perhaps include a comment saying why we want to ignore those tests.
A simple "We do not care about code formatting and git tests in building a package".
It sounds obvious, however I think it makes understanding easier.
If you feel strongly against this. Feel free to ignore.

**Comment by happysalada** [comment]
Looks good to merge. I just have a small comment about adding a comment describing why we ignore those tests.

**Pull request merged**